pkg/core/blockchain.go

```diff
@@ -1058,33 +1053,89 @@ func (bc *Blockchain) ForEachNEP17Transfer(acc util.Uint160, f func(*state.NEP17
 }
 
 // GetNEP17Balances returns NEP17 balances for the acc.
-func (bc *Blockchain) GetNEP17Balances(acc util.Uint160) *state.NEP17TransferInfo {
-	bs, err := bc.dao.GetNEP17TransferInfo(acc)
+func (bc *Blockchain) GetNEP17Balances(acc util.Uint160) (*state.NEP17Balances, error) {
+	result := state.NewNEP17Balances()
+	bw := io.NewBufBinWriter()
+	for _, h := range bc.contracts.Management.GetNEP17Contracts() {
+		cs, _ := bc.contracts.Management.GetContract(bc.dao, h) // no error can occur, as nep17 list is consistent with contract list
+		balance, err := bc.getNEP17Balance(h, acc, bw)
+		if err != nil {
+			return nil, err
```

Comment:
So if there is some ill-behaved contract (e.g. it just burns a lot of GAS in `balanceOf`), no info will ever be returned for _any_ account? I think we may safely skip this error.

Reply:
That's true, fixed.